### PR TITLE
Add number format expression

### PR DESCRIPTION
**Pull request by tobrun**
Closes #14840 

@chloekraw do we want to include this in `release-queso` or `release-ristretto`?

### CODE REVIEW

**Comment by chloekraw** [comment]
Thanks for the ping @tobrun. Let's release in `ristretto`. 

cc @maps-ios, we should try to get `number-format` bindings in `ristretto` as well.

**Comment by chloekraw** [comment]
Ah, sorry, Darwin bindings are already complete. I was confusing this with aggregated cluster properties.

**Comment by tobrun** [comment]
Adding `do not merge` label, waiting for branch cut of `release-queso`

File: platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/style/ExpressionTest.java [hunk 1]
```diff
@@ -561,6 +569,142 @@ public void testTextFieldFormattedArgument() {
...
+      Feature feature = Feature.fromGeometry(Point.fromLngLat(latLng.getLongitude(), latLng.getLatitude()));
+      feature.addNumberProperty("number_value", 12.345678);
+
+      mapboxMap.getStyle().addSource(new GeoJsonSource("source", feature));
+      SymbolLayer layer = new SymbolLayer("layer", "source");
+      mapboxMap.getStyle().addLayer(layer);
+
+      Expression numberFormatExpression = numberFormat(
+        number(get("number_value")),
+        locale("nl-BE"),
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/style/ExpressionTest.java]: NIT: Would be nice to have all of the options fetched from the source or use nested math expressions for the number of digits.

**Comment by chloekraw** [comment]
@tobrun before merging, could we add some more detail to the changelog about what the `number-format` expression does?

File: platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/style/ExpressionTest.java (hunk 1 above)
Comment (tobrun) [reply platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/style/ExpressionTest.java]: https://github.com/mapbox/mapbox-gl-native/pull/15424/commits/4562fe83d2e0e54f5d96b06764706844ebd86b46#diff-5da4e7ffcfa373969703b7ee7c12818bR689-R692

**Pull request merged**